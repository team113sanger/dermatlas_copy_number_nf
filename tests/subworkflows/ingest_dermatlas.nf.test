nextflow_workflow {

    name "Test Workflow DERMATLAS_METADATA"
    script "subworkflows/ingest_dermatlas.nf"
    workflow "DERMATLAS_METADATA"

    test("Should run without failures") {

        when {
            params {
                    "PROJECTDIR" = "/workspaces/ascat",
                    "OUTDIR" = "/workspaces/ascat/results"
            }
            workflow {
                """
                input[0] = file("tests/testdata/PD57536a.sample.dupmarked.bam")
                input[1] = file("tests/testdata/pair_ids.tsv")
                input[2] = file("tests/testdata/metadata_tab")
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
