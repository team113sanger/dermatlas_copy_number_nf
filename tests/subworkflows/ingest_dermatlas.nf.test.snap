{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    
                ],
                "combined_metadata": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-12T21:19:20.621978"
    }
}